packages/transformer/src/IModelTransformer.ts

```diff
@@ -2220,25 +2220,23 @@ export class IModelTransformer extends IModelExportHandler {
    * You generally should not call this function yourself and use [[process]] with [[IModelTransformOptions.argsForProcessChanges]] provided instead.
    * It is public for unsupported use cases of custom synchronization transforms.
    * @note if [[IModelTransformOptions.argsForProcessChanges]] are not defined in this transformation, this will fail
-   * without setting the `force` option to `true`
+   * without setting the `initializeReverseSyncVersion` option to `true`
+   * The initializeReverseSyncVersion is added to save the reverse synchronization version for forward synchronization transformations. When set to `true`, reverse sync version is saved
```

Comment:
```suggestion
   * The initializeReverseSyncVersion is added to set the reverse synchronization version during a forward synchronization. When set to `true`, the reverse sync version is saved as the current changeset of the targetDb. Note that typically the reverseSyncVersion is saved as the last changeset merged from the branch into master. A user might decide to pass true to avoid having a reverseSyncVersion of "". A reverseSyncVersion of "" interferes with a fix for a now fixed bug. That fix cannot be easily removed, so we must support the initialization of the reverseSyncVersion
   ```

Reply:
Not super in love with my suggestion but I would like to add more of the context for why this is necessaary.

Reply:
Probably it should also mention something about this only being needed for the initial transformation between a master and branch imodel

Reply:
@nick4598 which bug do you have in mind here? In our case, saving reverse synch version for forward synchronization is needed for partial fork workflow. If not saved, following transformation to synch new changes from fork to main iModel will fail, as transformer will process changes starting from first commit and will not remove deleted elements from main iModels. Send you more detailed documentation regarding this issue.
It is not needed to "+1" issue if you had that in mind.

Reply:
I responded on teams my understanding of the situation, but Ill copy and paste it here as well:

Maybe Daniel could provide some of the context, if we were to not persist the reverseSyncVersion then it would be "". When this happens the core transformer treats the cachedSyncVersion (same as `_syncrhoinzationVersion` I think) for the transformation as index -1 and id "".
 
 
```
// NOTE: that we do NOT download the changesummary for the last transformed version, we want
    // to ignore those already processed changes
    const startChangesetIndexOrId =
      args.startChangeset?.index ??
      args.startChangeset?.id ??
      this._synchronizationVersion.index + 1;
```
this makes the syncVersion index start at 0. I believe there is no changeset 0, atleast Daniel showed that to me that we do not download a changeset 0. This seems in line with what I remember as well, that changeset 0 is really just the initial seed imodel before any changesets.
 
The core transformer stores pendingReverseSyncIndices of 1 after the initial transform (note that this did not previously happen and was a bug that you guys ended up fixing by adding 1 to the reverseSyncVersion ). Since the core transformer stores that pendingreverseSyncIndice it skips the changeset that we need to skip. Its the one that brought over all the hcanges from the partial fork. Since you guys fixed the bug by adding 1 to reverseSyncVersion you need the reverseSYncVersion to be ;initialized; so that you have ;0 stored there instead of "". Your fix is to split on ; and add 1 to the version. Since this is the way you guys fixed it and it has implications for your database, we of course need to continue to support this.
 
Thats my understanding of it, maybe Ive got it wrong. 
 
Personally I find it a bit confusing that we are initializing the reverseSyncVersino because its a bit opposite to what the documentatino of reverseSyncVersion is whicvh is that its the 'last changeset index reverse synced into master'. Thats the main reason I wanted as much documentation surrounding this.

Reply:
Thanks for the context, Nick. I agree that adding the new initializeReverseSyncVersion parameter makes it confusing since this new behavior does not reflect what is documented. We have added an additional comment to let users know of the effects of using it when it is not the first transformations -   
   * `Note that typically, the reverseSyncVersion is saved as the last changeset merged from the branch into master.`
   * `Setting initializeReverseSyncVersion to true could overwrite this correct reverseSyncVersion and should only be done during the first transformation between a master and branch iModel.`

Reply:
Going to add this for sake of completeness: but flaw in my logic as julija pointed out is that I assume target iModel starts off at changeset 0. But there are some cases where they need to push up some schemas in a separate changeset to the target iModle before running transformation. Setting up the reverse sync version in those cases is required to skip the changeset that brought over schema changes.